testData.js

```diff
@@ -59,17 +63,24 @@ class TestData {
         if (testURL) {
             var response = await HTTP.getJSON(testURL);
             return new TestData(response, query);
-        }
-        else {
-            throw new Error("A 'url' parameter should be included on the query string.");
+        } else if (query.path) {
+            const content = await readFile(path.join(config.testsDir, query.path));
+            return new TestData(JSON.parse(content), query);
+        } else {
+            throw new Error("A 'url' or 'path' parameters should be included on the query string.");
         }
     }
 
     static async fromObject(obj, defaults) {
         var testData = null;
+        var response = null;

```

Comment:
why is `response` defined here?
unlike `testData` it is not used outside the scope of the `if/else` blocks, you can just define it using `const` in each block.
btw, in the `else` block, instead of `response` it's better names `parsedContent` or something similar... `response` implies a request

Reply:
fixed